orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/ExecutionLauncher.java

```diff
@@ -191,7 +190,8 @@ private PipelineExecution handleStartupFailure(PipelineExecution execution, Thro
     } else {
       log.error("Failed to start {} {}", execution.getType(), execution.getId(), failure);
     }
-    executionRepository.updateStatus(execution.getType(), execution.getId(), status);
+    execution.updateStatus(TERMINAL);
+    executionRepository.updateStatus(execution, TERMINAL);
```

Comment:
It is a little strange to have to chain these 2 calls together all the time. Are there known cases of people using Repositories beyond SQL or Redis where it is not possible to eliminate the old method and combine these in the repository implementation? That would require docs about the expectation also though.

Reply:
@Bijnagte you're right, it's a little awkward. I changed it in cc0dffc37 to make it more intuitive, it becomes:

```
execution.updateStatus(TERMINAL);
executionRepository.updateStatus(execution)
```

Removing the new status from the execution repository call disambiguates that the execution model is expected to have the new status.